net/FlatBuffers/ByteBuffer.cs

```diff
@@ -50,12 +54,16 @@ namespace FlatBuffers
 {
     public abstract class ByteBufferAllocator : IDisposable
```

Comment:
Should we remove IDisposable since the default implementation no longer needs it?  Custom allocators could still be disposable if needed.

Reply:
I think that could make sense... The caller that is creating the custom allocator can be responsible for disposing it.

Note that this means `ByteBuffer` would also not be `IDisposable` anymore.

Since the allocator isn't exposed on `ByteBuffer`, there would be no way you could create a custom/disposable allocator, pass it into a `ByteBuffer` and then hand that `ByteBuffer` off to some other code to dispose of it when they are done with it.

Do you think that scenario would be a problem?

Reply:
I don't think that scenario would be a problem.  It is not for any of mine.

Reply:
OK, I can remove `IDisposable` from both then. Any more feedback? Or will it be good after that change?